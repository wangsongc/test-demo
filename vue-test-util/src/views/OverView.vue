<template>
  <div>
    <h1 class="title">this is overview</h1>
    <filter-test></filter-test>
    <axios-test></axios-test>
    <router-test></router-test>
    <vuex-test></vuex-test>
  </div>
</template>
<script>
import FilterTest from '@/components/FilterAddWatchTest.vue'
import AxiosTest from '@/components/AxiosTest.vue'
import RouterTest from '@/components/RouterTest.vue'
import VuexTest from '@/components/VuexTest.vue'
export default {
  name: 'OverView',
  components: {
    FilterTest,
    AxiosTest,
    RouterTest,
    VuexTest
  },
  data () {
    return {
    }
  },
  methods: {
  }
}
</script>
<style scoped>
</style>
