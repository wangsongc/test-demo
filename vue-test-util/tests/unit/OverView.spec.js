/**
 * OverView.vue组件
 * 测试内容包括以下
 * 伪造子组件存根，当然也可以用真的子组件做存根，个人觉得没有必要
 */
import { shallowMount } from '@vue/test-utils'
import OverView from '@/views/OverView.vue'

describe('OverView.vue', () => {
  // A Wrapper is an object that contains a mounted component or vnode and methods to test the component or vnode
  // wrapper是一个对象，包含了已经mounted的组件或vnode，及测试该组件或 vnode 的方法。
  // wrapper.vm是组件实例，包含该实例的所有方法和属性
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(OverView, { stubs: ['filter-test', 'axios-test', 'router-test', 'vuex-test'] })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  // 测试内容：snapshot->概括的测试DOM结构
  // 如果组件内存在比较特殊的需要测试的DOM结构的话，可以单独测试（详见AppButton测试文件末尾）
  it('snapshot test', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
