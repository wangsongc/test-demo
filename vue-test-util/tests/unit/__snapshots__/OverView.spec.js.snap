// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverView.vue snapshot test 1`] = `
<div>
  <h1 class="title">this is overview</h1>
  <filter-test-stub></filter-test-stub>
  <axios-test-stub></axios-test-stub>
  <router-test-stub></router-test-stub>
  <vuex-test-stub></vuex-test-stub>
</div>
`;
